packages/server/lib/util/args.js

```diff
@@ -239,8 +240,16 @@ module.exports = {
 
     if (config) {
       // convert config to an object
-      // annd store the config
-      options.config = sanitizeAndConvertNestedArgs(config)
+      // and store the config
+      debug('parsing config:', config)
+      try {
+        options.config = sanitizeAndConvertNestedArgs(config)
+      } catch (err) {
+        debug('could not pass config %o', config)
+        debug('error %o', err)
+
+        return errors.throw('COULD_NOT_PARSE_ARGUMENTS', 'invalid config', config)
```

Comment:
i feel like this error should be thrown from within `sanitizeAndConvertNestedArgs`, and then we can apply it to `env` and `reporterOptions` as well, both use `sanitizeAndConvertNestedArgs`

Reply:
currently `--env foo` will return the same error we're preventing for `--config` here